charms/slurmctld/src/charm.py

```diff
@@ -391,7 +391,7 @@ def _resume_nodes(self, nodelist: List[str]) -> None:
         """Run scontrol to resume the specified node list."""
         nodes = ",".join(nodelist)
         update_cmd = f"update nodename={nodes} state=resume"
-        self._slurmctld.scontrol(update_cmd)
+        self._slurmctld.scontrol(*shlex.split(update_cmd))
```

Comment:
Is there a particular reason why we can't pass the command directly to `self._slurmctld.scontrol(...)`?

```suggestion
        self._slurmctld.scontrol("update", f"nodename={','.join(nodelist)}", "state=resume")
```

Typically I only use `shlex.split(...)` if there's shell syntax I want to preserve such as quoted values like `State=DOWN Reason="Scheduled maintenance"`:

```pycon
Python 3.12.3 (main, Nov  6 2024, 18:32:19) [GCC 13.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import shlex
>>> shlex.split("State=DOWN Reason='Scheduled maintenance'")
['State=DOWN', 'Reason=Scheduled maintenance']
```

Given that node names are continuous - e.g. names don't include whitespace characters - I think we can get away with just passing the arguments directly as strings here.